src/c-writer.cc

```diff
@@ -1082,17 +1100,27 @@ void CWriter::WriteDataInitializers() {
     Write("LOAD_DATA(", ExternalRef(memory->name), ", ");
     WriteInitExpr(data_segment->offset);
     Write(", data_segment_data_", data_segment_index, ", ",
-          data_segment->data.size(), ");", Newline());
+          data_segment->data.size());
+    Write(");", Newline());
     ++data_segment_index;
   }
 
   Write(CloseBrace(), Newline());
 }
 
 void CWriter::WriteElemInitializers() {
+  Write(Newline(), "static void init_table(void) ", OpenBrace());
+
+  if (!module_->types.size()) {
+    // If there are no function types there cannot be any table entries either.
```

Comment:
I don't think that's true in general. You can have a table that contains non-function types with wasm GC at least.

Reply:
Right, but wouldn't the mean that `module_->types` would have to be non-empty?  I'll update the comment.

Reply:
Ah, yes, if these are all types and not just function types, which I realize now must be the case, then just updating the comment sgtm.